        * timespec.h (timespec_cmp):
        Used direct comparison using if condition comparisons (i.e <, >,
        ==) to avoid integer overflow when we use subtraction.

--- timespec.h	2005-09-21 23:05:39.000000000 -0700
+++ tempTime.h	2017-10-27 18:21:45.973596000 -0700
@@ -45,8 +45,11 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  if (a.tv_sec < b.tv_sec) return -1;
+
+  else if (a.tv_sec > b.tv_sec) return 1;
+
+  return 0;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
